improvement publishr

boygirl opened the pull request:
cc/ @ryan-roemer does this look right to you?

boygirl committed:
add publishr scripts

ryan-roemer commented:
@boygirl -- I think you also need something like:

``` js
    "publishr": {
      "dependencies": ["^builder"]
    }
```

@jmcriffey -- Can we get your help / guidance / review on this one? First target repo is: https://github.com/FormidableLabs/victory-core/blob/master/package.json

I think we also may need something analogous to https://github.com/FormidableLabs/publishr/blob/master/.npmignore.publishr too (which will need to live in this archetype). Maybe something like adding:

``` js
    "publishr": {
      "dependencies": ["^builder"]
      "files": {
        ".npmignore": "node_modules/builder-victory-component/config/OTHER_DIR?/npmignore.publishr",
        ".someconfig": ".someconfig.publishr"
      }
    }
```

Target for  `victory-*` repo is:

```
lib/
dist/
demo/ # Maybe? @boygirl?
```

But within the above, we need to exclude:

```
dist/*.map
```

Also, 

Thanks!

boygirl commented:
@ryan-roemer I think the `publishr` dependency swapping needs to go in the individual repos, since it's acting on their `package.json`. Am I missing something?

ryan-roemer commented:
@boygirl -- Good point. Remove here and target `victory-*/package.json:devDependencies`.

boygirl commented:
@ryan-roemer `.npmignore` in `victory-*` is already _not_ ignoring `dist` and `lib`, and is ignoring `dist/*.map`. I think I can skip the `.npmignore.publishr` swapping in this case?

ryan-roemer commented:
@boygirl -- That sounds right. 👍

jeffrifwald commented on package.json:
```diff
@@ -16,6 +16,8 @@
     "npm:preversion": "builder run check",
     "npm:version": "builder run clean && builder run build",
+    "npm:postversion": "publishr postversion",
+    "npm:postpublish": "publishr postpublish && builder run clean",
```
This looks good. You might also want to include a command to contain `publishr dry-run` so you can catch errors before actually versioning or publishing.

boygirl committed:
add dry run script, remove unnecessary clean step on postpublish

boygirl commented:
Alright, I've added a dry-run script and removed `builder run clean` from `postpublish`. This is unnecessary now as `lib/` and `dist/` will be `.gitignore`'d

@ryan-roemer @jmcriffey good to merge?

jeffrifwald commented:
LGTM
